examples/basic_adding.py

```diff
@@ -24,162 +23,172 @@
+
+    def handle_order_placement(self, side: Side, book_data: Dict) -> None:
+        """Handle the placement and cancellation of orders based on the order book update."""
+        book_price = float(book_data["levels"][side_to_uint(side)][0]["px"])
+        ideal_distance = book_price * DEPTH
+        ideal_price = book_price + (ideal_distance * side_to_int(side))
+
+        provide_state = self.provide_state[side]
+
+        if provide_state["type"] == "resting":
+            self.maybe_cancel_order(side, provide_state, ideal_price, ideal_distance)
+        elif provide_state["type"] == "in_flight_order":
+            self.check_in_flight_order(side, provide_state)
+
+        if provide_state["type"] == "cancelled":
+            self.place_new_order(side, ideal_price)
+
+    def maybe_cancel_order(self, side: Side, provide_state: Resting, ideal_price: float, ideal_distance: float) -> None:
+        """Cancel the order if it deviates beyond the allowable limit."""
+        distance = abs(ideal_price - provide_state["px"])
+        if distance > ALLOWABLE_DEVIATION * ideal_distance:
+            oid = provide_state["oid"]
+            print(f"Cancelling order due to deviation: oid:{oid}, side:{side}, ideal_price:{ideal_price}")
+            response = self.exchange.cancel(COIN, oid)
+            if response["status"] == "ok":
+                self.recently_cancelled_oid_to_time[oid] = get_timestamp_ms()
+                self.provide_state[side] = {"type": "cancelled"}
+            else:
+                logging.error(f"Failed to cancel order {oid} for side {side}: {response}")
+
+    def check_in_flight_order(self, side: Side, provide_state: InFlightOrder) -> None:
+        """Check if the in-flight order has timed out."""
+        if get_timestamp_ms() - provide_state["time"] > ORDER_TIMEOUT:
+            print("Order is still in flight after timeout, treating as cancelled.")
+            self.provide_state[side] = {"type": "cancelled"}
+
+    def place_new_order(self, side: Side, ideal_price: float) -> None:
+        """Place a new order if conditions are met."""
+        if self.position is None:
+            logging.debug("Waiting for position refresh before placing order.")
+            return
+
+        size = MAX_POSITION + self.position * side_to_int(side)
+        if size * ideal_price < 10:
+            logging.debug("Order size too small, not placing order.")
+            return
+
+        px = float(f"{ideal_price:.5g}")
+        print(f"Placing order: size:{size}, price:{px}, side:{side}")
+        response = self.exchange.order(COIN, side == "B", size, px, {"limit": {"tif": "Alo"}})
+        if response["status"] == "ok":
+            status = response["response"]["data"]["statuses"][0]
+            if "resting" in status:
+                self.provide_state[side] = {"type": "resting", "px": px, "oid": status["resting"]["oid"]}
 
     def on_user_events(self, user_events: UserEventsMsg) -> None:
-        print(user_events)
```

Comment:
It feels inconsistent to log debug the book messages and not the user event messages